### DISCUSSION TITLE
Cloud connection instead of local

### DISCUSSION THREAD

**Discussion by Krautmaster** [petretiandrea/home-assistant-tapo-p100#682] [Ideas]
Well, obviously the future is cloud and or me like in any other normal household these plugs should be not within the same network. All smart home device in my place are in a guest network, not able to access my local devices. 

The normal apps are handling the same without issue though the cloud connection, no difference in using the products like lawn mowers, cleaning robots, hue, plugs and so on...  so this Tapo integration should habe two options, either have a local mode or a normal cloud mode where only login data is required. No IP data.

Does this work for now or will it ever? Is the public Tapo API closed and not able to be used by this integration?

**Comment by petretiandrea** [comment]
Home assistant was born with the purpose of being hub that can control devices in the absence of internet connection. It would be a shame not to be able to control devices when the internet is absent, so the integration is based on local control, which allows Tapo devices to be used offline. If you want to add remote devices, I recommend [HomeAssistant remote](https://github.com/custom-components/remote_homeassistant) or configure your network appropriately

**Comment by gerritgriebel** [comment]
Took a while to find this definite answer. I'm running HA in a docker image on my cloud server outside my home network and it works fine for all devices I need to control in two homes as they all have cloud integration with is supported by HA.  I have difficulties to understand why Tapo cloud integration needs the local IP of the device. The device could register itself by connecting tapo cloud service. 

Reading this, I did return my newly bought PT110 switches and looked for something that is supported by my cloud only setup. I ordered ANTELA WLAN Smart plug model F1s202 with Tuya integration instead and it works out of the box with no need for a HACS integration.

**Reply by MakaanPL** [reply]
Your design choices are totally the opposite of what a typical smart home user wants. Local control is the only way to ensure our devices will still work when internet connection fails or manufacturer stops supporting their products (due to business decision or e.g. bankruptcy).

**Reply by gerritgriebel** [reply]
Yes it's a different use case and it works fine for me with the set of devices I had/have when I discovered home assistent: a linux vm in the cloud, tado heating, two netatmo weather stations, two cloud controlled dehumidifiers and two tibber energy contracts with according metering. Just fireing up the docker image is the perfect setup for me.
